### PR TITLE
Head points & manual registration figure

### PR DESCRIPTION
**Pull request by Moo-Marc**
Edit: this was removed from this PR: {_As discussed in https://github.com/brainstorm-tools/brainstorm3/pull/567, this is an alternate fix to https://github.com/brainstorm-tools/brainstorm3/commit/422078ae45b6a2bf08352797218a812d55fc5f0b for head point coloring / colorbar behavior, where points are now registered as a figure surface._}

Remaining changes:

1. Minor fix to colormaps/SetMaxCustom: would crash if more than one surface. Fix copied from elsewhere in the code.
2. A "reset" button is added to the figure, to reset the registration.
3. Some checks are made on previous registration steps (manual/auto) and printed to the command window. This was added as a "passive" function (nothing saved, no output) in process_adjust_coordinates.

(Please ignore other changes in process_adjust_coordinates, mainly comment reformatting.)

### CODE REVIEW

**Comment by ftadel** [comment]
You asked me to work on these issues, I did it three days later. 
It took many hours of my work time, which is limited and quite packed.
I'm sorry, I'm not going to spend more time on this topic, for reviewing hundreds of lines of code that do something similar than what is already implemented in the master branch. 

If you need to add additional fixes, please build them on top of the current master branch.
Adding a new surface would have more side effects than what I proposed, I'm not in favor of it. Please do not open a PR that just projects this one on top of the current master branch.

Next time, please decide in advance if you want to work on something, or if you want me to work on it.
Don't do both as it is quite inefficient in terms of resource allocation.
If you are not sure what is the best approach, please ask me in advance, I'm easily available for short zoom meetings.

**Comment by Moo-Marc** [comment]
> If you are not sure what is the best approach, please ask me in advance

Sorry this was a case of miscommunication.  I guess my request was a bit vague, asking for advice and leaving the door open to more assistance, because I didn't know if you'd have time to work on this.  I started working on it while waiting to hearing back and making other changes as I was using the manual registration.  From now on, let's please both indicate when we're working on something.  And I'll be clearer in my requests.

The reset button is to reset the registration, not the figure view parameters (double-click).  It basically is a shortcut to closing without saving and reopening the manual registration figure.  I find it useful since there's no undo, e.g. if I press the wrong mouse button and make a big registration change when I meant to rotate the view.

I'll remove the points-as-surface changes and reopen this PR when I've finished testing.  The three other points are still useful, including a bug fix to SetMaxCustom if there's more than one surface.

**Comment by Moo-Marc** [comment]
I edited the initial post above to reflect the remaining changes in this PR.

The changes to panel_surface/GetSurface were to get the scalp surface specifically instead of the selected surface.  It seemed like a practical functionality to have - getting a surface by type.  If I missed a simpler way to do it, please let me know.

File: toolbox/core/bst_colormaps.m [hunk 1]
```diff
@@ -376,14 +376,26 @@ function SetMaxCustom(ColormapType, DisplayUnits, newMin, newMax)
...
                         % Get surfaces defined in this figure
                         TessInfo = getappdata(sFigure.hFigure, 'Surface');
-                        DataFig = TessInfo.DataMinMax;
-                        if ~isempty(TessInfo.DataSource.Type)
-                            DataType = TessInfo.DataSource.Type;
-                            isSLORETA = strcmpi(DataType, 'Source') && ~isempty(strfind(lower(TessInfo.DataSource.FileName), 'sloreta'));
-                            if isSLORETA 
-                                DataType = 'sLORETA';
+                        % Find surfaces that match this ColormapType
+                        iSurfaces = find(strcmpi({TessInfo.ColormapType}, ColormapType));
```
Comment (ftadel) [inline toolbox/core/bst_colormaps.m:380]: Getting the surface with the appropriate colormap is good.
Using 'EEG', 'MEG' or 'NIRS' as DataType is good too.

However, in which case do you have multiple surfaces with the same ColormapType?
If you can't produce this: please remove the loop on iSurfaces and only use the first one.

File: toolbox/process/functions/process_adjust_coordinates.m [hunk 2]
```diff
@@ -1,9 +1,8 @@
 function varargout = process_adjust_coordinates(varargin)
```
Comment (ftadel) [inline toolbox/process/functions/process_adjust_coordinates.m:1]: I did not review these changes.

File: toolbox/sensors/channel_align_manual.m [hunk 3]
```diff
@@ -403,10 +404,10 @@
 if isProgress
     bst_progress('stop');
 end
-    
-end
-
 
+% Check and print to command window if previously auto/manual registration, and if MRI fids updated.
+process_adjust_coordinates('CheckPrevAdjustments', in_bst_channel(ChannelFile), sMri);
```
Comment (ftadel) [inline toolbox/sensors/channel_align_manual.m:409]: I did not check this.

**Review by ftadel — COMMENTED** [review]
Looks good, just one little thing to check/fix.

File: toolbox/core/bst_colormaps.m (hunk 1 above)
Comment (Moo-Marc) [reply toolbox/core/bst_colormaps.m]: Done.

File: toolbox/process/functions/process_adjust_coordinates.m [hunk 4]
```diff
@@ -903,3 +855,79 @@
...
+    iAlign(iAlign < iImport) = [];
+    AlignType = 'none';
+    while ~isempty(iAlign)
+        % Check which adjustment was done last.
+        switch lower(ChannelMat.History{iAlign(end),3}(1:5))
+            case 'remov'
+                % Removed a previous step. Ignore corresponding adjustment and look again.
+                iAlign(end) = [];
+                if contains(ChannelMat.History{iAlign(end),3}, 'AdjustedNative')
+                    iAlignRemoved = find(contains(ChannelMat.History(iAlign,3), 'AdjustedNative'), 1, 'last');
```
Comment (ftadel) [inline toolbox/process/functions/process_adjust_coordinates.m:882]: Contains was added in 2016b: must be removed.

File: toolbox/process/functions/process_adjust_coordinates.m [hunk 5]
```diff
@@ -903,3 +855,79 @@
...
+    % Flag if auto or manual registration performed, and if MRI fids updated. Print to command
+    % window for now.
+    if any(~isfield(ChannelMat, {'History', 'HeadPoints'}))
+        % Nothing to check.
+        return;
+    end
+    if nargin < 2 || isempty(sMri) || ~isfield(sMri, 'History')
+        iMriHist = [];
+    else
+        iMriHist = find(strcmpi(sMri.History(:,3), 'Applied digitized anatomical fiducials'), 1, 'last');
```
Comment (ftadel) [inline toolbox/process/functions/process_adjust_coordinates.m:868]: Where is this defined?

File: toolbox/process/functions/process_adjust_coordinates.m [hunk 6]
```diff
@@ -903,3 +855,79 @@
...
+            case 'refin'
+                % Automatic MRI-points alignment
+                AlignType = 'auto';
+                break;
+            case 'align'
+                % Manual MRI-points alignment
+                AlignType = 'manual';
+                break;
+        end
+    end
```
Comment (ftadel) [inline toolbox/process/functions/process_adjust_coordinates.m:907]: Where is all this coming from?

File: toolbox/sensors/channel_align_manual.m (hunk 3 above)
Comment (ftadel) [reply toolbox/sensors/channel_align_manual.m:409]: I'm not sure I understand what is happening in this function.
I am not comfortable adding this here, as this is a very standard function.

If the idea is only to display the previous changes done on the file: you can simply right-click on the channel file > File > View history.

**Review by ftadel — CHANGES_REQUESTED** [review]

File: toolbox/process/functions/process_adjust_coordinates.m (hunk 6 above)
Comment (Moo-Marc) [reply toolbox/process/functions/process_adjust_coordinates.m]: removed

**Review by ftadel — APPROVED** [review]

**Pull request merged**